src/database/DataTypes/ItemBatch.js

```diff
@@ -145,6 +150,20 @@ export class ItemBatch extends Realm.Object {
 
     return newLocationId !== currentLocationId;
   }
+
+  applyLocation(database, newLocation) {
+    if (!this.shouldApplyLocation(newLocation)) return null;
```

Comment:
Hm, feels like maybe this should be called by the caller rather than here? I feel like if the calling code tells the object to do something, the object shouldn't decide it knows better? I get that it probably enables cleaner calling code, but it feels like it might be a bit of an anti-pattern?

Reply:
💯 1f2e2fed2c06fec822d90d19c9f791c6259616f7